rest_framework/schemas/openapi.py

```diff
@@ -554,7 +554,7 @@ def map_field_validators(self, field, schema):
             if isinstance(v, URLValidator):
                 schema['format'] = 'uri'
             if isinstance(v, RegexValidator):
-                schema['pattern'] = v.regex.pattern
+                schema['pattern'] = v.regex.pattern.replace('\\Z', '\\z')
```

Comment:
I think a comment here. 

```suggestion
                # In Python, the token \Z does what \z does in other engines.
                # https://stackoverflow.com/questions/53283160
                schema['pattern'] = v.regex.pattern.replace('\\Z', '\\z')
```